quipucords/tests/api/scanjob/test_scanjob.py

```diff
@@ -1997,3 +1984,46 @@ def test_filter_by_status(self, django_client, mocker):
         result = response.json()["results"][0]
         assert result["id"] == scanjob.id
         assert result["status"] == chosen_status
+
+
+@pytest.fixture
+def scanjob_with_inspect_results():
+    """ScanJob with 3 InspectResults."""
+    inspection_results = InspectResultFactory.create_batch(3)
+    scanjob = ScanJobFactory()
+    task = ScanTaskFactory(job=scanjob)
+    task.inspect_results.add(*inspection_results)
+    return scanjob
+
+
+@pytest.mark.django_db
+def test_delete_inspect_results(scanjob_with_inspect_results):
+    """Test ScanJob.delete_inspect_results method."""
+    scanjob = scanjob_with_inspect_results
+    assert InspectResult.objects.filter(tasks__job=scanjob).exists()
+    scanjob.delete_inspect_results()
+    assert not InspectResult.objects.filter(tasks__job=scanjob).exists()
+
+
+@pytest.mark.django_db
+def test_delete_inspect_results_binded_to_other_scanjobs(
+    scanjob_with_inspect_results,
+):
+    """Ensure only InspectResults uniquely bound to a ScanJob are deleted."""
+    inspect_res = InspectResult.objects.first()
+    other_scanjob = ScanJobFactory()
+    task = ScanTaskFactory(job=other_scanjob)
+    inspect_res.tasks.add(task)
+    # save id in a variable so we can check later for existence
+    inspect_res_id = inspect_res.id
+
+    scanjob = scanjob_with_inspect_results
+    scanjob.delete_inspect_results()
```

Comment:
```suggestion
    scanjob = scanjob_with_inspect_results
    assert InspectResult.objects.filter(tasks__job=scanjob).count() == 3
    scanjob.delete_inspect_results()
```

As I was reworking that ugly delete queryset, I found myself doubting the state of the objects, and I even opened a text file to keep track of the ids of which objects were related to each other. If we have any doubt about the delete query, for maximum readability, we might want to assert a few extra things like this before and after the deletes so that future readers can better intuit the state of things.

Not a must-have. Just a passing thought I had as I was working through unfamiliar code.

Reply:
Sorry, github auto resolved this when I accepted the suggestion.